dex/src/main/scala/com/wavesplatform/dex/api/websockets/actors/WsHandlerActor.scala

```diff
@@ -0,0 +1,155 @@
+                        Behaviors.same
+                      }
+                  }
+
+                case subscribe: WsOrderBookSubscribe =>
+                  // TODO DEX-700 test for order book that hasn't been created before
+                  if (subscribe.depth <= 0) clientRef ! WsError.from(error.RequestArgumentInvalid("depth"), time.getTimestamp())
+                  else
+                    assetPairBuilder.validateAssetPair(subscribe.key).value.onComplete {
+                      case Success(Left(e)) => clientRef ! WsError.from(e, time.getTimestamp())
+                      case Success(Right(_)) =>
+                        matcherRef ! MatcherActor.AggregatedOrderBookEnvelope(
+                          subscribe.key,
+                          AggregatedOrderBookActor.Command.AddWsSubscription(clientRef)
+                        )
+                      case Failure(e) =>
+                        context.log.warn(s"An error during validation the asset pair ${subscribe.key}", e)
+                        clientRef ! WsError.from(error.WavesNodeConnectionBroken, time.getTimestamp())
+                    }
+                  Behaviors.same
+
+                case subscribe: WsAddressSubscribe =>
+                  subscribe.validate(settings.jwtPublicKey, AddressScheme.current.chainId) match {
+                    case Left(e) =>
+                      context.log.debug(s"WsAddressSubscribe(k=${subscribe.key}, t=${subscribe.authType}) failed with ${e.code}")
+                      clientRef ! WsError.from(e, time.getTimestamp())
+                    case Right(_) =>
+                      context.log.debug(s"WsAddressSubscribe(k=${subscribe.key}, t=${subscribe.authType}) is successful")
+                      addressRef ! AddressDirectory.Envelope(subscribe.key, AddressActor.WsCommand.AddWsSubscription(clientRef))
+                  }
+                  Behaviors.same
+
+                case unsubscribe: WsUnsubscribe =>
+                  unsubscribe.key match {
+                    case Inl(x) =>
+                      context.log.debug(s"WsUnsubscribe(assetPair=$x)")
+                      matcherRef ! MatcherActor.AggregatedOrderBookEnvelope(x, AggregatedOrderBookActor.Command.RemoveWsSubscription(clientRef))
+                    case Inr(Inl(x)) =>
+                      context.log.debug(s"WsUnsubscribe(address=$x)")
+                      addressRef ! AddressDirectory.Envelope(x, AddressActor.WsCommand.RemoveWsSubscription(clientRef))
+                    case Inr(Inr(_)) =>
+                  }
+                  Behaviors.same
+              }
+
+            case command: Command.ProcessClientError =>
+              context.log.debug("Got an error from the client, stopping: {}", command.error)
+              Behaviors.stopped
+
+            case Completed =>
+              List(nextPing, pongTimeout, maxLifetimeExceeded, firstPing).foreach { _.cancel() }
```

Comment:
Here and in line 138 - can be extracted in separate `def`

Reply:
Done